core/src/test/scala/kafka/server/KafkaRequestHandlerTest.scala

```diff
@@ -53,7 +53,7 @@ class KafkaRequestHandlerTest {
     val context = new RequestContext(requestHeader, "0", mock(classOf[InetAddress]), new KafkaPrincipal("", ""),
       new ListenerName(""), SecurityProtocol.PLAINTEXT, mock(classOf[ClientInformation]), false)
     val request = new RequestChannel.Request(0, context, time.nanoseconds(),
-      mock(classOf[MemoryPool]), mock(classOf[ByteBuffer]), metrics)
+      mock(classOf[MemoryPool]), ByteBuffer.allocate(0), metrics)
```

Comment:
It's not possible to mock this class with Java 20, but there's no reason to mock it.